### ISSUE TITLE
reward using custom dataset

### ISSUE THREAD

**Issue by yxk9810** [pat-jj/DeepRetrieval#21]
![Image](https://github.com/user-attachments/assets/8fc9792f-9e7d-45aa-a250-ab157602492b)

I trained using my own data with the Qwen2.5-3B model. For retrieval, I used a domain-finetuned BGE model and employed the top-1 retrieval accuracy as the reward. The training dataset consists of over 1,000 samples, and the test dataset contains 100 samples. However, as shown in the figure, the reward does not steadily increase but fluctuates continuously. Could you provide suggestions on potential optimizations or areas for improvement?

**Comment by pat-jj** [comment]
Thanks for raising the issue! What type of data is it? Our study reveals that if the data's distribution has been seen by the dense retriever, the room left for query rewriting would be very limited.

If the data was not been used for retriever's pre-training, here's my suggestion:
1. Try to adjust the reward to top-k retrieval acc instead of only top-1, which can let the model learn the rewriting strategy easier.
2. Try BM25. Our result reveals that BM25+DeepRetrieval can outperform dense retrieval on several datasets.

**Comment by yxk9810** [comment]
On our data, BM25 performs much worse than BGE—by more than 20 points—so we chose BGE.

On our training data, reinforcement learning doesn't seem suitable. The top-1 recall is already around 52%, and even after modifying the reward to focus on top-5, the results are still quite unstable. For top-10, the rewards might all be positive. Could it be that our task is too simple?

Are you using the original Qwen-3B model or a fine-tuned version?

**Comment by yxk9810** [comment]
We trained for 10 epochs, but the results remained the same. Maybe the training data is too limited, or the task itself is too simple
It's quite difficult to collect real-world data.I'll go ahead and close this issue for now, and later we’ll see if model improvements or expanding the dataset can lead to better results.
